torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -163,6 +194,23 @@ void ProcessGroupNCCL::WorkNCCL::synchronize() {
       AT_CUDA_CHECK(cudaDeviceSynchronize());
     }
   }
+
+  // In case of blocking, wait for the operation to complete.
+  if (blockingWait_) {
```

Comment:
similar code might need to be added to `AsyncAllreduceCUDAWork` in `ProcessGroupGloo`. It can come in a separate PR.